staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -89,12 +91,22 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 		Proxy: c.Proxy,
 	}
 
-	if c.ExecProvider != nil && c.AuthProvider != nil {
+	if c.Exec.ExecProvider != nil && c.AuthProvider != nil {
 		return nil, errors.New("execProvider and authProvider cannot be used in combination")
 	}
 
-	if c.ExecProvider != nil {
-		provider, err := exec.GetAuthenticator(c.ExecProvider)
+	if c.Exec.ExecProvider != nil {
+		caData, err := dataFromSliceOrFile(c.CAData, c.CAFile)
```

Comment:
for a CAFile config, this reads/freezes the data from the file at TransportConfig() construction time rather than transport construction time... probably not a big deal, but notable... could end up with different content here than is used in the transport itself.

Reply:
it's also maybe slightly unexpected that TransportConfig() would do file reads...

Reply:
I could pass in the logic as a closure to delay the computation if you think the extra complexity would be worth it?